### PR TITLE
enable building with clang

### PR DESCRIPTION
**Pull request by khilman**
Add support to the build tools and docker images to support clang

Prerequisites to merge:
- new docker images are build/pushed to docker hub
- the jenkins jobs need updates for the new variables: COMPILER changed to CC, and CC_VERSION was added.

### CODE REVIEW

**Comment by khilman** [comment]
Tested on staging with CC=clang CC_VERSION=8:
https://staging-ci.kernelci.org/job/khilman/job/kernel-build/24/console

**Comment by khilman** [comment]
Tested on staging with CC=gcc CC_VERSION=7:
https://staging-ci.kernelci.org/job/khilman/job/kernel-build/25/console

File: jenkins/build.jpl [hunk 1]
```diff
@@ -109,7 +111,8 @@ node("docker && builder") {
     Describe:  ${params.GIT_DESCRIBE}
     Revision:  ${params.COMMIT_ID}
     Config:    ${params.DEFCONFIG}
-    Compiler:  ${params.COMPILER}
+    C Compiler: ${params.CC}
```
Comment (gctucker) [inline jenkins/build.jpl]: How about this instead:
```
Compiler:  ${params.CC} ${params.CC_VERSION}
```
to keep it easy to read on a single line?

File: jenkins/dockerfiles/clang-8/Dockerfile [hunk 2]
```diff
@@ -7,4 +7,7 @@ RUN wget -q -O - https://apt.llvm.org/llvm-snapshot.gpg.key| apt-key add -
 RUN apt-add-repository 'deb http://apt.llvm.org/unstable/ llvm-toolchain main'
 RUN apt-get update && apt-get install --no-install-recommends -y \
     clang-8 clang
+RUN apt-get autoremove -y gcc
 
+# clang kernel builds still use arch-specific GNU as/ld
+RUN apt-get -y install binutils-aarch64-linux-gnu
```
[the remark this answers is not in the document]
Comment (khilman) [reply jenkins/dockerfiles/clang-8/Dockerfile]: OK

File: jenkins/dockerfiles/clang-8/Dockerfile [hunk 3]
```diff
@@ -7,4 +7,7 @@ RUN wget -q -O - https://apt.llvm.org/llvm-snapshot.gpg.key| apt-key add -
 RUN apt-add-repository 'deb http://apt.llvm.org/unstable/ llvm-toolchain main'
 RUN apt-get update && apt-get install --no-install-recommends -y \
```
[the remark this answers is not in the document]
Comment (khilman) [reply jenkins/dockerfiles/clang-8/Dockerfile]: OK, I've fixed this now.

File: jenkins/build.jpl (hunk 1 above)
Comment (khilman) [reply jenkins/build.jpl]: OK

File: src/org/kernelci/util/Job.groovy [hunk 4]
```diff
@@ -34,17 +34,21 @@ def addBoolParams(params, bool_params) {
...
+
     if (arch == "riscv")
         docker_arch = "riscv64"
     else if ((arch == "i386") || (arch == "x86_64"))
         docker_arch = "x86"
     else
         docker_arch = arch
 
-    return "${base}${compiler}_${docker_arch}"
+    return "${base}${cc}-${cc_version}_${docker_arch}"
```
[the remark this answers is not in the document]
Comment (khilman) [reply src/org/kernelci/util/Job.groovy]: OK, I've done that cleanup/reorg and also fixed the kernel_arch vs arch bug.

**Comment by khilman** [comment]
OK, I updated this branch to to include all the changes requested.  
Tested on staging with clang-8:
https://staging-ci.kernelci.org/job/khilman/job/kernel-build/31/console

Tested on staging with gcc-7:
https://staging-ci.kernelci.org/job/khilman/job/kernel-build/30/console

**Comment by gctucker** [comment]
The issue above is actually more likely to be due to #70.

**Pull request merged**